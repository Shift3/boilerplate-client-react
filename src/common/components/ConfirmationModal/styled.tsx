import { FC } from 'react';
import Button, { ButtonProps } from 'react-bootstrap/Button';
import Modal, { ModalProps } from 'react-bootstrap/Modal';
import styled from 'styled-components';

// Based on the styled-components docs at https://styled-components.com/docs/api#caveat-with-function-components,
// in order for typechecking to work correctly with styled components that extend a function components, we need
// to define the component and it's type first as done here.
const BootstrapModal: FC<ModalProps> = ({ children, ...rest }) => <Modal {...rest}>{children}</Modal>;
const BootstrapButton: FC<ButtonProps> = ({ children, ...rest }) => <Button {...rest}>{children}</Button>;

export const StyledModal = styled(BootstrapModal)`
  & .modal-header {
    border: none;
  }

  & .modal-content {
    padding: 50px;
    background-color: ${(props) => props.theme.primary};
    border-color: ${(props) => props.theme.primary};
  }

  & .modal-title {
    color: ${(props) => props.theme.cardHeader};
  }

  & .modal-body {
    padding: 10px 0;
    color: white;
  }

  & .modal-footer {
    margin-top: 50px;
    padding: 0;
    border: none;
    justify-content: center;
  }

  button:first-of-type {
    margin-right: 10px;
  }
`;

export const StyledCancelButton = styled(BootstrapButton)`
  min-width: 146px;
  min-height: 24px;
  padding: 10px;
  margin: 0;

  &,
  &:hover,
  &:active,
  &:focus {
    background-color: ${(props) => props.theme.primary} !important;
    border-color: ${(props) => props.theme.primaryBorder} !important;
  }

  &:focus {
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgb(52 58 64 / 50%);
  }
`;

export const StyledConfirmButton = styled(BootstrapButton)`
  min-width: 146px;
  min-height: 24px;
  padding: 10px;
  margin: 0;

  &,
  &:hover,
  &:active,
  &:focus {
    background-color: ${(props) => props.theme.accent} !important;
    border-color: ${(props) => props.theme.accentBorder} !important;
  }

  &:focus {
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgb(52 58 64 / 50%);
  }
`;
