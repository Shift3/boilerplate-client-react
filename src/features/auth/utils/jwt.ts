import jwtDecode, { JwtPayload } from 'jwt-decode';

export const isMalformed = (token: string): boolean => {
  try {
    jwtDecode<JwtPayload>(token);
  } catch (e) {
    return true;
  }
  return false;
};

export const isExpired = (token: string): boolean => {
  const decoded = jwtDecode<JwtPayload>(token);
  const { exp } = decoded;
  // eslint-disable-next-line no-magic-numbers
  return !!exp && exp * 1000 < Date.now();
};