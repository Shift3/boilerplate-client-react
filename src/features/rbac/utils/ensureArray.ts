export const ensureArray = <T>(elem: T | T[]): T[] => (Array.isArray(elem) ? elem : [elem]);
