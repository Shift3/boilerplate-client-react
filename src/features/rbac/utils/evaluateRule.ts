import { User } from 'common/models';
import { Rule } from '../rules';

export const evaluateRule = (rule: Rule, user: User, data?: unknown): boolean => {
  if (typeof rule === 'function') {
    return rule(user, data);
  }

  return rule;
};
