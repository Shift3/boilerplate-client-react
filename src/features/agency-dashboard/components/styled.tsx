import { FC } from 'react';
import Button, { ButtonProps } from 'react-bootstrap/Button';
import styled from 'styled-components';

const BootstrapButton: FC<ButtonProps> = ({ children, ...rest }) => <Button {...rest}>{children}</Button>;

export const StyledFormWrapper = styled.div`
  max-width: 500px;
  min-width: 500px;
  padding: 50px;
  background-color: ${(props) => props.theme.primary};
  border-radius: 0.25rem;

  & label {
    color: white;
  }

  & .invalid-feedback {
    /* Modify style of validation error messages here */
  }
`;

export const StyledFormTitle = styled.p`
  color: ${(props) => props.theme.cardHeader};
  font-size: 2em;
  font-weight: 500;
`;

export const StyledCancelButton = styled(BootstrapButton)`
  min-width: 146px;
  min-height: 24px;
  padding: 10px;
  margin: 0;

  &,
  &:hover,
  &:active,
  &:focus {
    background-color: ${(props) => props.theme.primary} !important;
    border-color: ${(props) => props.theme.primaryBorder} !important;
  }

  &:focus {
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgb(52 58 64 / 50%);
  }
`;

export const StyledFormButtonWrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 50px;

  & button {
    margin-right: 10px;
  }

  & button:last-of-type {
    margin-right: 0;
  }
`;

export const StyledSubmitButton = styled(BootstrapButton)`
  min-width: 146px;
  min-height: 24px;
  padding: 10px;
  margin: 0;

  &,
  &:hover,
  &:active,
  &:focus {
    background-color: ${(props) => props.theme.accent} !important;
    border-color: ${(props) => props.theme.accentBorder} !important;
  }

  &:focus {
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgb(52 58 64 / 50%);
  }
`;
