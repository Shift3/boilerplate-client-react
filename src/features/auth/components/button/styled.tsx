import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const CustomButton = styled(Button)`
  background-color: ${(props) => props.theme.buttons.backgroundColor};
  border: none;
  color: ${(props) => props.theme.buttons.text};

  width: ${(props) => props.theme.buttons.width};
  &:hover {
    background-color: ${(props) => props.theme.buttons.hoverBackgroundColor};
    /* Leaving these style values comments for reference
    border-color: ${(props) => props.theme.buttons.hoverBorderColor}; */
  }
  &:disabled {
    background-color: ${(props) => props.theme.buttons.disabledBackgroundColor};
    /* Leaving these style values comments for reference
    border-color: ${(props) => props.theme.disabledBorderColor}; */
  }
`;
