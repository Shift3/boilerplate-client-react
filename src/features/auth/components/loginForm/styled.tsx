import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import styled from 'styled-components';

export const Login = styled.div`
  padding: 20px;
`;

export const FormLabel = styled(Form.Label)`
  color: ${(props) => props.theme.forms.label};
  margin-bottom: 5px;
`;

export const FormControl = styled(Form.Control)`
  color: ${(props) => props.theme.forms.control};
  margin-bottom: 20px;
`;

export const InputError = styled.span`
  color: ${(props) => props.theme.forms.inputError};
`;

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-top: 16px;
`;

export const CancelButton = styled(Button)`
  color: ${(props) => props.theme.forms.white};
  background-color: ${(props) => props.theme.forms.buttonColor};
  min-width: 180px;
  min-height: 24px;
`;

export const LogInButton = styled(Button)`
  color: ${(props) => props.theme.forms.white};
  background-color: ${(props) => props.theme.forms.buttonColor};
  border: none;
  min-width: 180px;
  min-height: 24px;
`;
