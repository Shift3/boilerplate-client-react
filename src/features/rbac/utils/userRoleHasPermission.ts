import { User } from 'common/models';
import { Permission, RbacRules } from '../rules';
import { evaluateRule } from './evaluateRule';

export const userRoleHasPermission = (
  rules: RbacRules,
  permission: Permission,
  user: User,
  data?: unknown,
): boolean => {
  const permissionMap = rules[user.role];

  if (!permissionMap) {
    return false;
  }

  const rule = permissionMap[permission];

  if (!rule) {
    return false;
  }

  return evaluateRule(rule, user, data);
};
