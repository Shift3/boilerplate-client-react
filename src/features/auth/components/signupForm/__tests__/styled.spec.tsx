import renderer from 'react-test-renderer';
import { expectToMatchSnapshot } from 'utils/test';
import { StyledForm, Title, FormLabel, InputError, ButtonWrapper, CancelButton, SignUpButton } from '../styled';

import 'jest-styled-components';

describe('<SignUpForm /> styled components', () => {
  it('should match the stored <StyledForm /> snapshot', () => expectToMatchSnapshot(<StyledForm />));

  it('should match the stored <Title /> snapshot', () => expectToMatchSnapshot(<Title />));

  it('should match the stored <FormLabel /> snapshot', () => expectToMatchSnapshot(<FormLabel />));

  it('should match the stored <InputError /> snapshot', () => expectToMatchSnapshot(<InputError />));

  it('should match the stored <ButtonWrapper /> snapshot', () => expectToMatchSnapshot(<ButtonWrapper />));

  it('should match the stored <CancelButton /> snapshot', () =>
    expect(renderer.create(<CancelButton />).toJSON()).toMatchSnapshot());

  it('should match the stored <SignUpButton /> snapshot', () =>
    expect(renderer.create(<SignUpButton />).toJSON()).toMatchSnapshot());
});
