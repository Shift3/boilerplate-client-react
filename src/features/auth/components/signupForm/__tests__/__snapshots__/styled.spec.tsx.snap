// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> styled components should match the stored <ButtonWrapper /> snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 10px;
}

<div
  className="c0"
/>
`;

exports[`<SignUpForm /> styled components should match the stored <CancelButton /> snapshot 1`] = `
.c0 {
  color: white;
  border: 1px solid #679daa;
  margin-right: 20px;
}

<button
  className="c0 btn btn-primary"
  disabled={false}
  type="button"
/>
`;

exports[`<SignUpForm /> styled components should match the stored <FormLabel /> snapshot 1`] = `
.c0 {
  color: white;
}

<label
  className="c0 form-label"
/>
`;

exports[`<SignUpForm /> styled components should match the stored <InputError /> snapshot 1`] = `
.c0 {
  color: lightgrey;
}

<span
  className="c0"
/>
`;

exports[`<SignUpForm /> styled components should match the stored <SignUpButton /> snapshot 1`] = `
.c0 {
  color: white;
}

<button
  className="c0 btn btn-primary"
  disabled={false}
  type="button"
/>
`;

exports[`<SignUpForm /> styled components should match the stored <StyledForm /> snapshot 1`] = `
.c0 {
  padding: 40px;
  border-radius: 5px;
  width: 400px;
}

<form
  className="c0"
/>
`;

exports[`<SignUpForm /> styled components should match the stored <Title /> snapshot 1`] = `
.c0 {
  font-size: 2em;
  font-style: bold;
  padding-bottom: 10px;
}

<div
  className="c0"
/>
`;
